### DISCUSSION TITLE
DeprecationWarning: Call to __init__()

### DISCUSSION THREAD

**Discussion by Philip-Leron** [flet-dev/flet#3370] [Q&A]
DeprecationWarning: Call to __init__() is deprecated since version 0.21.0 and will be removed in version 1.0. UserControl is deprecated. See https://flet.dev/docs/getting-started/custom-controls.
  super().__init__()

**Comment by Philip-Leron** [comment]
meanwhile i cant see anything in the https://flet.dev/docs/getting-started/custom-controls and my app is not running too

**Comment by ndonkoHenri** [comment]
Please fill up all details as required by the issue template.
As the message says, `UserControl` is deprecated and will be removed in the future.

**Comment by Philip-Leron** [comment]
error is coming fro this class Main(UserControl):
    def __init__(self, page:Page,):
        super().__init__().. code was working until i upgraded flet... old flet version doent have some properties i required....and this error shows up when i try running my app

**Comment by Philip-Leron** [comment]
this is what i have......
 from pages.forgotpassword import ForgotPassword
from pages.login import Login
from pages.signup import SignUp
from pages.home import Home
from flet import *
class Main(UserControl):
    def __init__(self, page:Page):
        super().__init__()
        self.page = page
        self.page.title="Portal"
        self.page.horizontal_alignment=MainAxisAlignment.CENTER
        self.page.vertical_alignment=CrossAxisAlignment.CENTER
        #self.page.alignment=MainAxisAlignment.CENTER
        self.page.window_width=1800
        self.page.window_height=1000
        self.page.window_center=True
        self.page.padding=30
        
        self.init_helper()
    def init_helper(self,):
        self.page.on_route_change = self.on_route_change
        self.page.go('/')
        
    def on_route_change(self, route):
        newpage ={
            "/":Home,
            "/login":Login,
            "/signup":SignUp,
            "/forgotpassword":ForgotPassword
            

        }[self.page.route](self.page)
        self.page.views.clear()
        self.page.views.append(
            View(
                route,
                [newpage]

            )

        )


app(target=Main,assets_dir='assets')

**Comment by ndonkoHenri** [comment]
Please build some reproducible code so we can test.

**Comment by Philip-Leron** [comment]
https://github.com/Philip-Leron/flet_portal

**Comment by ndonkoHenri** [comment]
> https://github.com/Philip-Leron/flet_portal

The repo has no python files.

**Reply by Philip-Leron** [reply]
repo is updated

**Comment by rlimaeco** [comment]
I have the same warning, but my app still working , there is any other alternative to UserControl ? it would be nice to have more information  on the warning or in the link that show in the warning

**Reply by Philip-Leron** [reply]
my app doesn't work at all

**Comment by ndonkoHenri** [comment]
usercontrol will be working normally in the next flet release.
the alternative to the deprecated usercontrol: https://flet.dev/docs/getting-started/custom-controls
